truffleruby/src/main/ruby/core/file.rb

```diff
@@ -1348,9 +1348,30 @@ def world_writable?
 end
 
 STDIN = File.new(0)
+
+def STDIN.read(length, outbuf=nil)
+  bytes = Truffle::Interop.polyglot_in_read length
+  return nil unless bytes
+  if outbuf
+    outbuf << bytes
+    outbuf
+  else
+    bytes
+  end
+end
+
 STDOUT = File.new(1)
+
+def STDOUT.write(bytes)
+  Truffle::Interop.polyglot_out_write bytes
+end
+
 STDERR = File.new(2)
 
+def STDERR.write(bytes)
+  Truffle::Interop.polyglot_err_write bytes
+end
```

Comment:
Yes, but it's the only way to support Polyglot streams.
I doubt this guard matter, it's just a comparison compared to a syscall writing a buffer.
It's also hiding less the logic which is a great for debug-ability.